lib/src/ui/movies_page.dart

```diff
@@ -14,11 +18,225 @@ class _MoviesPageState extends State<MoviesPage> {
+                        color: Colors.grey,
+                        fontFamily: 'Play-Bold'),
+                  ),
+                ),
+              ),
+              StreamBuilder(
+                stream: _moviesBloc.swiper,
+                builder: (
+                  context,
+                  AsyncSnapshot<Movies> snapshot,
+                ) {
+                  return snapshot.hasData
+                      ? _swiper(
+                          snapshot,
+                        )
+                      : Center(
+                          child: Padding(
+                            padding: const EdgeInsets.only(
+                                top: UiConstants
+                                    .swiperProgressIndicatorTopPadding),
+                            child: CircularProgressIndicator(
+                              strokeWidth:
+                                  UiConstants.progressIndicatorStrokeWidth,
+                              backgroundColor: Colors.purple,
+                              valueColor: new AlwaysStoppedAnimation<Color>(
+                                  Colors.black),
+                            ),
+                          ),
+                        );
+                },
+              ),
+              StreamBuilder(
+                stream: _moviesBloc.stream,
+                builder: (
+                  context,
+                  AsyncSnapshot<Movies> snapshot,
+                ) {
+                  return snapshot.hasData
+                      ? buildList(
+                          snapshot,
+                        )
+                      : Padding(
+                          padding: const EdgeInsets.only(
+                              top: UiConstants.gridProgressIndicatorTopPadding),
+                          child: Center(
+                            child: CircularProgressIndicator(
+                              strokeWidth:
+                                  UiConstants.progressIndicatorStrokeWidth,
+                              backgroundColor: Colors.purple,
+                              valueColor: new AlwaysStoppedAnimation<Color>(
+                                  Colors.black),
+                            ),
+                          ),
+                        );
+                },
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget _swiper(AsyncSnapshot snapshot) {
+    return Container(
+        height: UiConstants.swiperCardsHeight,
+        child: Container(
+          child: Swiper(
+            itemBuilder: (BuildContext context, int index) {
+              return new Image.network(
+                snapshot.data.results[index].posterPath,
+                fit: BoxFit.fill,
+              );
+            },
+            itemCount: snapshot.data.results.length,
+            itemWidth: UiConstants.swiperCardsWidth,
+            layout: SwiperLayout.STACK,
+          ),
+        ));
+  }
```

Comment:
You can abstract the swiper widget into a separate file.

Reply:
done